### DISCUSSION TITLE
serversideup/php:8.2-fpm-nginx installing PHP 8.3?

### DISCUSSION THREAD

**Discussion by Offlein** [serversideup/docker-php#223] [Q&A] [answered]
### Affected Docker Images

serversideup/php:8.2-fpm-nginx

### Current Behavior

My deploy has just broken because -- assuming I'm reading this correctly; I could be an idiot -- it's trying to install PHP 8.3 and associated libraries.

Last week the same automated process ran without mention of PHP 8.3.

### Expected Behavior

Everything should be PHP 8.2.

### Steps To Reproduce

My automated deploy runs with a[n unchanged] Dockerfile:

```
FROM serversideup/php:8.2-fpm-nginx as production
EXPOSE 80
ARG USERID
ARG GROUPID
ENV SSL_MODE=off
WORKDIR /var/www/html
COPY --chown=${USERID:-1000}:${GROUPID:-1000} . .
VOLUME [ "/var/www/html/storage" ]
RUN apt-get update  \
    && apt-get install -y php-imagick php-redis
RUN composer install --no-interaction --optimize-autoloader --no-dev

FROM production as dev
RUN apt-get install -y php-xdebug
```

I have an automated process that builds that dockerfile and deploys it.

### Host Operating System

Github Actions "ubuntu-latest" runner

### Docker Version

```markdown
Unknown
```

**Comment by jaydrogers** [comment, marked as answer]
### Not a bug with `serversideup/php`
I ran a quick test and we're shipping PHP 8.2 by default still (had me worried 😅)
![Screenshot 2023-11-27 at 14 07 50@2x](https://github.com/serversideup/docker-php/assets/3174134/6be83727-1339-4ffd-8949-86d8d8d320f9)

### Possible culprit
You have these lines:
```
apt-get install -y php-imagick php-redis
```
Change them to:
```
apt-get install -y php8.2-imagick php8.2-redis
```

Let me know if this works!

**Reply by Offlein** [reply]
@jaydrogers, thank you. I should've thought to try that. I do remember not specifying 8.2 for the packages when I wrote them, but my logic was "It's an 8.2 image, so it'll be 8.2 packages, and if we upgrade, it's better to have explicitly specified fewer things." I assume serversideup is based off Ubuntu or something and they've simply bumped the default version (?). 

It really doesn't matter too much to me, since I'm now specifying, but it feels as though it might be "more proper" to lock the php8.2-xx-xx image default PHP packages to be 8.2? Probably is a whole big can of worms, but might be a vector of confusion  for others.

Anyway, for me I'm fixed -- thank you! I'll get back to inserting my own bugs into my production code.

**Reply by jaydrogers** [reply]
> It really doesn't matter too much to me, since I'm now specifying, but it feels as though it might be "more proper" to lock the php8.2-xx-xx image default PHP packages to be 8.2? 

That behavior is dictated by the Ondrej PHP repo: https://github.com/oerdnj/deb.sury.org

We will be moving away from this behavior in the v3 of these images. **This will include breaking changes.** Be sure to subscribe to our releases for latest updates or on our mailing list: https://serversideup.net/subscribe/

More on our plan here: https://www.youtube.com/watch?v=JKR-zwwaJ1A

**Reply by Offlein** [reply]
Got it -- thanks so much!

**Reply by ssatz** [reply]
Thanks you so much! i was struggling this with for past 3 hrs. You saved me finally
